tests/rust/tcp-test/close/async_close.rs

```diff
@@ -0,0 +1,218 @@
+        Ok(_) => anyhow::bail!("wait() should succeed with async_close()"),
+        Err(_) => anyhow::bail!("wait() should succeed with async_close()"),
+    }
+
+    Ok(())
+}
+
+/// Attempts to close a TCP socket that is bound.
+fn async_close_bound_socket(libos: &mut LibOS, local: &SocketAddrV4) -> Result<()> {
+    println!("{}", stringify!(async_close_bound_socket));
+
+    // Create a bound socket.
+    let sockqd: QDesc = libos.socket(AF_INET, SOCK_STREAM, 0)?;
+    libos.bind(sockqd, *local)?;
+
+    // Succeed to close socket.
+    let qt: QToken = libos.async_close(sockqd)?;
+
+    // Poll once to ensure the async_close() coroutine runs and finishes the close.
+    match libos.wait(qt, Some(Duration::from_micros(0))) {
+        Ok(qresult) if qresult.qr_opcode == demi_opcode_t::DEMI_OPC_CLOSE => {},
+        Ok(_) => anyhow::bail!("wait() should succeed with async_close()"),
+        Err(_) => anyhow::bail!("wait() should succeed with async_close()"),
+    }
+
+    Ok(())
+}
+
+/// Attempts to close a TCP socket that is listening.
+fn async_close_listening_socket(libos: &mut LibOS, local: &SocketAddrV4) -> Result<()> {
+    println!("{}", stringify!(async_close_listening_socket));
+
+    // Create a listening socket.
+    let sockqd: QDesc = libos.socket(AF_INET, SOCK_STREAM, 0)?;
+    libos.bind(sockqd, *local)?;
+    libos.listen(sockqd, 16)?;
+
+    // Succeed to close socket.
+    let qt: QToken = libos.async_close(sockqd)?;
+
+    // Poll once to ensure the async_close() coroutine runs and finishes the close.
+    match libos.wait(qt, Some(Duration::from_micros(0))) {
+        Ok(qresult) if qresult.qr_opcode == demi_opcode_t::DEMI_OPC_CLOSE => {},
+        Ok(_) => anyhow::bail!("wait() should succeed with async_close()"),
+        Err(_) => anyhow::bail!("wait() should succeed with async_close()"),
+    }
+
+    Ok(())
+}
+
+/// Attempts to close a TCP socket that is accepting.
+fn async_close_accepting_socket(libos: &mut LibOS, local: &SocketAddrV4) -> Result<()> {
+    println!("{}", stringify!(async_close_accepting_socket));
+
+    // Create an accepting socket.
+    let sockqd: QDesc = libos.socket(AF_INET, SOCK_STREAM, 0)?;
+    libos.bind(sockqd, *local)?;
+    libos.listen(sockqd, 16)?;
+    let qt: QToken = libos.accept(sockqd)?;
+
+    // Poll once to ensure that the accept() co-routine runs.
+    match libos.wait(qt, Some(Duration::from_micros(0))) {
+        Err(e) if e.errno == libc::ETIMEDOUT => {},
+        Ok(_) => anyhow::bail!("wait() should not succeed with accept()"),
+        Err(_) => anyhow::bail!("wait() should timeout with accept()"),
+    }
+
+    // Succeed to close socket.
+    let qt_close: QToken = libos.async_close(sockqd)?;
+
+    // Poll once to ensure the async_close() coroutine runs and finishes the close.
+    match libos.wait(qt_close, Some(Duration::from_micros(0))) {
+        Ok(qresult) if qresult.qr_opcode == demi_opcode_t::DEMI_OPC_CLOSE => {},
```

Comment:
Could you rename `qresult` to `qr` so that it follows the convetions on the source tree?